users/models.py

```diff
@@ -101,6 +101,18 @@ class InterfaceType(models.TextChoices):
         choices=settings.LANGUAGES,
     )
 
+    class ApiAccessTier(models.TextChoices):
+        FREE = "free", "Free"
+        PRO = "pro", "Pro"
+        ENTERPRISE = "enterprise", "Enterprise"
```

Comment:
The whole concept looks good to me. 
I would suggest renaming these to represent the level of the access that is closer to the code and less to whatever business/product name we might have since we are not even sure we'll have those. Something long the lines of: LIMITED, STANDARD, FULL, or RESTRICTED, EXTENDED, UNRESTRICTED, or something along these lines.